src/System.Memory/src/System/Buffers/BuffersExtensions.cs

```diff
@@ -139,6 +139,10 @@ private static void WriteMultiSegment<T>(IBufferWriter<T> writer, in ReadOnlySpa
                 if (input.Length > 0)
                 {
                     destination = writer.GetSpan();
+
+                    if (destination.IsEmpty)
```

Comment:
nit: Can you please add a `Debug.Assert(writeSize != 0)` after the line:
```C#
int writeSize = Math.Min(destination.Length, input.Length);
```

Reply:
I think that it would make the tests `WritingToTooSmallSingleSegmentBufferFailsWithException` and `WritingToTooSmallMultiSegmentBufferFailsWithException`  fail in Debug:


```cs
int writeSize = Math.Min(destination.Length, input.Length);

Debug.Assert(writeSize != 0); // would throw here

if (input.Length > 0)
    if (destination.IsEmpty)
        ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.writer); // instead of here
```

And the `WritingEmptyBufferToMultipleSegmentEmptyBufferWriterDoesNothing` would fail instead of passing too (now it enters the loop and quits after the first iteration, with the assert it would throw):

```cs
while (true)
{
    int writeSize = Math.Min(destination.Length, input.Length);
    input.Slice(0, writeSize).CopyTo(destination); // does nothing for empty spans

    if (input.Length > 0) // not true for an empty span
    {
        
    }

    return; // returns after first iteration
}
```